pkg/server/server.go

```diff
@@ -934,9 +938,11 @@ func (s *ImmuServer) LoadDatabase(ctx context.Context, req *schema.LoadDatabaseR
 
 	s.dbList.Put(db)
 
-	if dbOpts.isReplicatorRequired() {
-		err = s.startReplicationFor(db, dbOpts)
-		if err != nil && err != ErrReplicatorNotNeeded {
+	err = s.startReplicationFor(db, dbOpts)
+	if err != nil {
+		if err == ErrReplicatorNotNeeded {
```

Comment:
same as above, probably the logging about not needed can be done inside the `startReplicationFor` method? so it will be only in one place